### DISCUSSION TITLE
Conditions Option for Customfield visibility

### DISCUSSION THREAD

**Discussion by Zombie-Toad** [netbox-community/netbox#18164] [General]
Hi everyone,

I created some custom field for different device roles (hardware and software version for switches, driver name for printer etc.). To reduce the clutter of input fields in the device form I asked myself if it would be possible to add some kind of filter. 
I don't know what Django is capable of so I need to ask.

Would it be possible to add a conditions field to customfields, just like the conditions field in "Event Rules"?
![Conditions](https://github.com/user-attachments/assets/5ad6630e-2d54-4068-9679-399aafde6e41)
As a result the input field only shows up if the condition matches the assigned role.
If there could be a way to do this, I would create a feature request.

If there is any other idea to achieve this (besides custom scripts, I already created some to create devices), please let me know.

**Comment by candlerb** [comment]
Netbox does not support altering the data input form (whether for custom fields or normal fields) based on attributes of the object. Sorry.  You would have to write a Plugin.

**Reply by Zombie-Toad** [reply]
OK thanks for the quick reply. Maybe we live with the big form, writing a plugin is a little bit too much for my little python knowledge.
